### ISSUE TITLE
[react-aira: RadioGroup]: No tabindex/Can't keyboard focus to input until value is checked

### ISSUE THREAD

**Issue by lukethacoder** [adobe/react-spectrum#5747]
### Provide a general summary of the issue here

When using the Controlled Value example, you are unable to keyboard focus to the input to select options. 

Once you have checked an item using the mouse, the input allows for keyboard focus and interactivity

### 🤔 Expected Behavior?

I should be able to keyboard navigate to the input

### 😯 Current Behavior

I cannot keyboard navigate to the input

### 💁 Possible Solution

Without diving into the source code, I assume there is an initial tabIndex that is not being set?

### 🔦 Context

Trying to use the Controlled Value example in a build that requires a high level of a11y.

### 🖥️ Steps to Reproduce

Example is live on the current documentation site:

https://react-spectrum.adobe.com/react-aria/RadioGroup.html#controlled-value

<img width="100%" alt="image" src="https://github.com/adobe/react-spectrum/assets/13529535/ce3db87c-e0ea-4d45-9f09-d2e53e560277">

### Version

react-aria: 3.31.1 react-aria-components: 1.0.1

### What browsers are you seeing the problem on?

Firefox, Chrome, Microsoft Edge

### What operating system are you using?

Windows 11

**Comment by yihuiliao** [comment]
Looks like it broke in this PR (https://github.com/adobe/react-spectrum/pull/4985). The logic regarding how the tab indexes are set will probably need to revisited.

**Comment by majornista** [comment]
Looks like the default selected value in the example is an empty string, which is not the same as `null`. https://github.com/adobe/react-spectrum/pull/4985/files#diff-26ddea7c3bd0c2b48179af98635706a6391a01b59c936e8b497d1843a97ff202R71-R78

**Comment by yihuiliao** [comment]
Oh yep, changing it to null seems to fix that, I can submit a quick PR for that

**Comment by federicocappellotto97** [comment]
I have the same problem
 "react-aria": "^3.32.1",
"react-aria-components": "^1.1.1",

**Comment by LFDanLu** [comment]
@federicocappellotto97 Mind sharing a code sandbox or reproduction? Note that the change was a docs specific one so you'll have to do the same kind of change for your implementation

**Comment by federicocappellotto97** [comment]
> @federicocappellotto97 Mind sharing a code sandbox or reproduction? Note that the change was a docs specific one so you'll have to do the same kind of change for your implementation

This is the most ugly reproduction i've ever done ahah [sandbox](https://codesandbox.io/p/devbox/gallant-gianmarco-72pdlq?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsp0zkan00073b6hu72aqz4o%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsp0zkan00023b6h9yby5jz1%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsp0zkan00043b6h8iu7jc3a%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsp0zkan00063b6hccig69o2%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsp0zkan00023b6h9yby5jz1%2522%253A%257B%2522id%2522%253A%2522clsp0zkan00023b6h9yby5jz1%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clsp0zkan00063b6hccig69o2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsp0zkan00053b6h5qwrddlf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clsp0zkan00063b6hccig69o2%2522%252C%2522activeTabId%2522%253A%2522clsp0zkan00053b6h5qwrddlf%2522%257D%252C%2522clsp0zkan00043b6h8iu7jc3a%2522%253A%257B%2522id%2522%253A%2522clsp0zkan00043b6h8iu7jc3a%2522%252C%2522activeTabId%2522%253A%2522clsp1lf1003qu3b6h73yflxcw%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522install%2522%252C%2522id%2522%253A%2522clsp1l8gg03op3b6hhh2l5fgb%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clsp1lf1003qu3b6h73yflxcw%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) but you can see that the radios have tab-index -1, so i can't focus them

**Comment by LFDanLu** [comment]
odd, looks like the radio has tabindex of 0 here: 
<img width="1243" alt="image" src="https://github.com/adobe/react-spectrum/assets/9661127/26b6e191-8006-49c9-ac91-1aecf5427ea8">

**Comment by federicocappellotto97** [comment]
> sandbox

Ok, i think i found the problem. You can try again [sandbox](https://codesandbox.io/p/devbox/gallant-gianmarco-72pdlq?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsp0zkan00073b6hu72aqz4o%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsp0zkan00023b6h9yby5jz1%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsp0zkan00043b6h8iu7jc3a%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsp0zkan00063b6hccig69o2%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsp0zkan00023b6h9yby5jz1%2522%253A%257B%2522id%2522%253A%2522clsp0zkan00023b6h9yby5jz1%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clsp0zkan00063b6hccig69o2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsp0zkan00053b6h5qwrddlf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clsp0zkan00063b6hccig69o2%2522%252C%2522activeTabId%2522%253A%2522clsp0zkan00053b6h5qwrddlf%2522%257D%252C%2522clsp0zkan00043b6h8iu7jc3a%2522%253A%257B%2522id%2522%253A%2522clsp0zkan00043b6h8iu7jc3a%2522%252C%2522activeTabId%2522%253A%2522clspzetp001dn3b6h861syp92%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522install%2522%252C%2522id%2522%253A%2522clspzdul9014k3b6hnbvj4cmp%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clspzetp001dn3b6h861syp92%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

It looks when you put defaultValue="" tabIndexes go to -1, and i'n my project since i'm using react-hook-form that's my default value if you don't specify it

**Comment by LFDanLu** [comment]
I see, in that case for now you'll have to specify the defaultValue to null like we do in the docs. We'll want to eventually handle the case where the value provided doesn't exist in the radiogroup

**Comment by Gregoor** [comment]
I'm facing the same issue with a controlled input (coming from `react-aria-cyetomponents`). I'm also setting an empty string, but for `value` not `defaultValue` and am still expecting focus to land on the first radio (where in reality it skips over the radios)

**Comment by snowystinger** [comment]
@Gregoor, I think you want to set the `value` to `null` for no value selected. Empty string is a valid value, so unless one of your options has that, I'm not sure what it'd do.

**Comment by Gregoor** [comment]
@snowystinger I did try that as well, but same result. Also a type error, since the props for `RadioGroup` are typed with `ValueBase<string>` which disallows null.

**Comment by snowystinger** [comment]
Ah, that'll be fixed in the next release `null` now https://github.com/adobe/react-spectrum/blob/main/packages/%40react-types/radio/src/index.d.ts#L32
https://github.com/adobe/react-spectrum/commit/bf7071fd3a9dbf8ec63d2c4767f44216f1eeca7c

**Comment by Gregoor** [comment]
Huh, and when I try repro-ing it, [it does indeed work with value set to null](https://codesandbox.io/p/devbox/aria-radio-t374pz). Then I'm doing something wrong (and I have a clue what, I am using a custom `useFocusRing`, for my JS-based styling system). Thanks!
